api/app/src/external/commonwell/document/document-query.ts

```diff
@@ -183,31 +184,45 @@ async function dowloadAndUploadDocsToS3({
 
           const data = await promise;
 
-          return {
+          const docWithFile: DocumentWithFilename = {
             ...doc,
             content: {
               ...doc.content,
               location: data.Location,
             },
             fileName: data.Key,
           };
-        }
 
-        return undefined;
+          const FHIRDocRef = toFHIRDocRef(docWithFile, organization, patient);
+          await upsertDocumentToFHIRServer(FHIRDocRef);
+        }
       } catch (error) {
         capture.error(error, {
           extra: {
             context: `s3.documentUpload`,
           },
         });
         throw error;
+      } finally {
+        // TODO: eventually we will have to update this to support multiple HIEs
+        try {
+          const newPatient = await getPatientOrFail({ id: patient.id, cxId: patient.cxId });
+
+          await updateDocQuery({
+            patient: newPatient,
+            status: "processing",
+            progress: {
+              completed: completedCount + 1,
+              total: documents.length,
+            },
+          });
+        } catch (err) {
+          capture.error(err, {
+            extra: { context: `cw.getDocuments.updateDocQuery` },
+          });
+        }
+        completedCount = completedCount + 1;
```

Comment:
will this be an issue if 2 promises try to read/update the patient at the same time?

Reply:
I dont believe so since we are locking when updating